ios/PresentationLayer/Sources/PresentationLayer/Utilities/BaseViewModel.swift

```diff
@@ -18,4 +21,21 @@ class BaseViewModel {
     deinit {
         Logger.info("%@ deinitialized", "\(type(of: self))", category: .lifecycle)
     }
+    
+    /// Override this method in a subclass for custom behavior when a view appears
+    public func onAppear() {}
+    
+    /// Override this method in a subclass for custom behavior when a view disappears
+    public func onDisappear() {
+        // Cancel all tasks when we are going away
+        tasks.forEach { $0.cancel() }
+    }
+    
+    func executeTask(_ task: Task<Void, Never>) -> Task<Void, Never> {
+        tasks.append(task)
+        return Task {
+            await task.value
+            tasks = tasks.filter { $0 != task } // Remove task when done
```

Comment:
Nemuze tady dojit k nejake conflict concurrency erroru? 🤔 Je to toto prirazeni blokujici? Resp kdyz tam najednou v 1 chvili pristoupi 2 tasky, tak zda-li si to nebudou menit pod rukama? Nemuze tady dojit k tomu, ze kdyz tam najednou budou 2 tasky, tak si ve stejnou chvili "porovnavaji" stejne listy a pak z nich kazdy vyfiltruje ten svuj prvek, ale ten ten z toho druheho tam stale bude a kdyz se to zpetne priradi do `tasks` tak tam bude ten, ktery "prohral"?

Reply:
"Should not happen" ™️ 😄

Simple synchronization was added in the last commit.
https://stackoverflow.com/questions/24045895/what-is-the-swift-equivalent-to-objective-cs-synchronized
https://www.cocoawithlove.com/blog/2016/06/02/threads-and-mutexes.html